patches/tModLoader/Terraria/ModLoader/UI/ModBrowser/UIModBrowser.cs

```diff
@@ -23,26 +48,44 @@ internal partial class UIModBrowser : UIState, IHaveBackButtonCommand
 	public UIModDownloadItem SelectedItem;
 
 	// TODO maybe we can refactor this as a "BrowserState" enum
-	public bool Loading;
+	public bool Loading => !ModList.State.IsFinished();
 	public bool anEnabledModUpdated;
 	public bool aDisabledModUpdated;
 	public bool aNewModDownloaded;
 
+	private bool _firstLoad = true;
+
 	private bool _updateAvailable;
 	private string _updateText;
 	private string _updateUrl;
 	private string _autoUpdateUrl;
 	private string _specialModPackFilterTitle;
-	private List<string> _specialModPackFilter;
+	private List<ModPubId_t> _specialModPackFilter;
 	private readonly List<string> _missingMods = new List<string>();
 
-	private readonly List<UIModDownloadItem> _items = new List<UIModDownloadItem>();
+	private HashSet<string> modSlugsToUpdateInstallInfo = new();
 
+	public const int DEBOUNCE_MS = 100;
```

Comment:
Need a comment on why we debounce. Better name `MinTimeBetweenUpdates`. Use `TimeSpan`